src/master-playlist-controller.js

```diff
@@ -852,42 +853,36 @@ export class MasterPlaylistController extends videojs.EventTarget {
    * @return {TimeRange} the current time
    */
   setCurrentTime(currentTime) {
-    let buffered = Ranges.findRange(this.tech_.buffered(), currentTime);
+    const media = this.masterPlaylistLoader_ && this.masterPlaylistLoader_.media();
+    const segments = media && media.segments || [];
 
-    if (!(this.masterPlaylistLoader_ && this.masterPlaylistLoader_.media())) {
-      // return immediately if the metadata is not ready yet
-      return 0;
+    // Don't start segment loaders at the new location if:
+    // * master media is not ready
+    // * master media has no segments
+    if (!media || !segments.length) {
```

Comment:
This likely doesn't exit early if we have a masterPlaylistLoader but no media loaded yet. Potentially why the tests are failing.